src/pke/include/encoding/plaintext.h

```diff
@@ -112,9 +113,10 @@ class PlaintextImpl {
   size_t level;
   size_t depth;
   usint slots;
+  EncryptionTechnique encTech;
 
  public:
-  PlaintextImpl(std::shared_ptr<Poly::Params> vp, EncodingParams ep,
+  PlaintextImpl(std::shared_ptr<Poly::Params> vp, EncodingParams ep, EncryptionTechnique encTech = STANDARD,
```

Comment:
should we combine all these constructors in to a templated one?

Reply:
Each function has a different typeFlag: IsPoly, IsDCRTPoly, and IsNativePoly. Do you think it's worth templating it and then type checking?

Reply:
Discussed with Dmitriy.